syntax = "proto3";

message NetworkRecord {
    int32 ID = 1;
    int32 pkSeqID = 2;
    double stime = 3;
    string flgs = 4;
    int32 flgs_number = 5;
    string proto = 6;
    int32 proto_number = 7;
    string saddr = 8;
    int32 sport = 9;
    string daddr = 10;
    int32 dport = 11;
    int32 pkts = 12;
    int32 bytes = 13;
    string state = 14;
    int32 state_number = 15;
    double ltime = 16;
    int32 seq = 17;
    double dur = 18;
    double mean = 19;
    double stddev = 20;
    double sum = 21;
    double min = 22;
    double max = 23;
    int32 spkts = 24;
    int32 dpkts = 25;
    int32 sbytes = 26;
    int32 dbytes = 27;
    double rate = 28;
    double srate = 29;
    double drate = 30;
    int32 TnBPSrcIP = 31;
    int32 TnBPDstIP = 32;
    int32 TnP_PSrcIP = 33;
    int32 TnP_PDstIP = 34;
    int32 TnP_PerProto = 35;
    int32 TnP_Per_Dport = 36;
    double AR_P_Proto_P_SrcIP = 37;
    double AR_P_Proto_P_DstIP = 38;
    int32 N_IN_Conn_P_DstIP = 39;
    int32 N_IN_Conn_P_SrcIP = 40;
    double AR_P_Proto_P_Sport = 41;
    double AR_P_Proto_P_Dport = 42;
    double Pkts_P_State_P_Protocol_P_DestIP = 43;
    double Pkts_P_State_P_Protocol_P_SrcIP = 44;
    int32 attack = 45;
    string category = 46;
    string subcategory = 47;
}
